pageserver/src/bin/pageserver.rs

```diff
@@ -502,6 +596,7 @@ fn start_pageserver(
                 "Got {}. Terminating gracefully in fast shutdown mode",
                 signal.name()
             );
+            shutdown.take();
```

Comment:
I think the `shutdown_pageserver` and `shutdown` names are misleading.
At this point, the `shutdown_pageserver` cancellation tree is only used for `wait_and_report`.

If you plan to replace the task_mgr shutdown token over time with `shutdown_pageserver` subtrees, at a mimimum, please make it clear in a comment.
Either here, or at the init site of the shutdown token.

I'd also be happy to just rename `shutdown_pageserver` to `cancel_wait_and_report`.

```suggestion
            // This cancels the `shutdown_pageserver` cancellation tree.
            // Right now that tree doesn't reach very far, and `task_mgr` is used instead.
            // The plan is to change that over time.
            shutdown.take();
```

Reply:
Hmm I guess I was initially thinking of naming this `shutdown_pageserver` as well, but yes, messy and so on. And perhaps I was thinking of it more like a signal "pageserver is now shutting down", but channels when named other than `(_tx, _rx)` frequently have this issue :)

In the final form, would probably just race signals:

```rust
let pageserver = start_pageserver(...).await;

tokio::select! {
  _ = sigquit() => { /* same std::process::exit as now */ },
  _ = sigint() => {},
  _ = sigterm() => {},
}

shutdown_pageserver.take();

let exit = pageserver.wait_for_shutdown().await;

std::process::exit(exit);
```

But I'll change the name of binding and include your comment.

Reply:
8cd28410